storage/backup/backup-cli/src/metadata/mod.rs

```diff
@@ -58,6 +58,86 @@ impl Metadata {
         })
     }
 
+    pub fn compact_epoch_ending_backup_range(
+        backup_metas: Vec<EpochEndingBackupMeta>,
+    ) -> Result<(Vec<TextLine>, ShellSafeName)> {
+        ensure!(
+            !backup_metas.is_empty(),
+            "compacting an empty metadata vector"
+        );
+        let backup_meta = backup_metas[0].clone();
+        let first_epoch = backup_meta.first_epoch;
+        let mut next_epoch = backup_meta.last_epoch + 1; // non inclusive
+        let mut res = Vec::new();
+        res.push(Metadata::EpochEndingBackup(backup_meta).to_text_line()?);
+        for backup in backup_metas.iter().skip(1) {
+            ensure!(
+                next_epoch == backup.first_epoch,
+                "Epoch ending backup ranges is not continuous expecting epoch {}, got {}",
+                next_epoch,
+                backup.first_epoch,
+            );
+            next_epoch = backup.last_epoch + 1;
+            res.push(Metadata::EpochEndingBackup(backup.clone()).to_text_line()?)
+        }
+        let name = format!(
+            "epoch_ending_compacted_{}-{}.meta",
```

Comment:
when you are running it the second time, you will be overwriting the files themselves? And then when you "backup" the files, you are cleanning the metadata folder?

(shall we add a test to compact a backup repeatedly?)